From c57524986e1af5bc816befa5e4285e314187ee44 Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Wed, 17 May 2023 11:43:30 +0200
Subject: [PATCH] target/linux-5.15: Add support for Neoway N75-EA

Serial driver option is missing IDs for Neoway N75-EA, so adding it and
thus making this card work.
---
 .../backport-5.15/999-neoway-n75-ea-support.patch    | 12 ++++++++++++
 1 file changed, 12 insertions(+)
 create mode 100644 target/linux/generic/backport-5.15/999-neoway-n75-ea-support.patch

diff --git a/target/linux/generic/backport-5.15/999-neoway-n75-ea-support.patch b/target/linux/generic/backport-5.15/999-neoway-n75-ea-support.patch
new file mode 100644
index 0000000000..bfa864e99a
--- /dev/null
+++ b/target/linux/generic/backport-5.15/999-neoway-n75-ea-support.patch
@@ -0,0 +1,12 @@
+diff --git a/drivers/usb/serial/option.c b/drivers/usb/serial/option.c
+index 6b69d05e2fb0..927196dedc8d 100644
+--- a/drivers/usb/serial/option.c
++++ b/drivers/usb/serial/option.c
+@@ -612,6 +612,7 @@ static void option_instat_callback(struct urb *urb);
+ 
+ 
+ static const struct usb_device_id option_ids[] = {
++	{ USB_DEVICE(0x2949, 0x8247), .driver_info = RSVD(4) },
+ 	{ USB_DEVICE(OPTION_VENDOR_ID, OPTION_PRODUCT_COLT) },
+ 	{ USB_DEVICE(OPTION_VENDOR_ID, OPTION_PRODUCT_RICOLA) },
+ 	{ USB_DEVICE(OPTION_VENDOR_ID, OPTION_PRODUCT_RICOLA_LIGHT) },
-- 
2.40.1

